### PR TITLE
[COMPLIANCE] Add copyright and license headers

### PR DESCRIPTION
**Pull request by bendbennett**
Closes #93 

This PR introduces:
- modification to `make generate` command that will auto-generate documentation and all copyright headers for this repo (hooks into `go generate`)
- GH workflow that will error if copyright headers aren't present or if license is malformed
- Addition to the `CONTRIBUTING.md` on how to run license autogenerating commands
- Add new `copywrite` module and add to dependabot

### CODE REVIEW

File: .github/workflows/compliance.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+jobs:
+  # Reference: ENGSRV-059
+  copywrite:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - uses: hashicorp/setup-copywrite@867a1a2a064a0626db322392806428f7dc59cb3e # v1.1.2
+      - run: copywrite headers --plan
+      - run: copywrite license
```
Comment (austinvalle) [inline .github/workflows/compliance.yml:17]:
```suggestion
      - run: copywrite license --plan
```
I believe I had this incorrect! This actually needs a `--plan` flag as well to throw an error!

File: main.go
[the remark this answers is not in the document]
Comment (bendbennett) [reply main.go:1]: I've just switched to using separate directories for `copywrite` and `tools`. Seems to be behaving as desired.

[the remark this answers is not in the document]
Comment (austinvalle) [reply main.go:1]: Cool 👍🏻, I'll follow suit in `terraform-plugin-go`

**Review by austinvalle — APPROVED** [review]

**Review by bflad — APPROVED** [review]
Looks great to me 🚀 Excited for cleaner Go modules. 🧹

**Pull request merged**